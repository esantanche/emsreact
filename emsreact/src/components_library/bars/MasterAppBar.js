import React, { Component } from 'react';
import PropTypes from 'prop-types';

// Material UI components
import AppBar from '@material-ui/core/AppBar';

// Material UI uses withStyles to inject an array of styles into the DOM as CSS, using JSS
import { withStyles } from "@material-ui/core/styles/index";

// This array of style gets injected into the component as "classes"
const styles = {
    MasterAppBar: {
        borderRadius: "4px",
    }
};

/**
 * The AppBar in the header. Its children are the home button and the menu button.
 *
 * @param {object} classes Classes injected by withStyles function
 */
class MasterAppBar extends Component {

    static propTypes = {
        classes: PropTypes.object.isRequired
    };

    render() {

        const { classes } = this.props;

        return (

            <AppBar position="sticky" className={classes.MasterAppBar}>

                {this.props.children}

            </AppBar>

        )
    }

}

export default withStyles(styles)(MasterAppBar);

