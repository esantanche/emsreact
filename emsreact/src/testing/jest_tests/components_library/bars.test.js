// Here we test components in components_library/bars

import React from "react";
import renderer from 'react-test-renderer';

import MasterAppBar from "../../../components_library/bars/MasterAppBar";

it('MasterAppBar renders correctly', () => {

    const tree = renderer
        .create(<MasterAppBar />)
        .toJSON();
    expect(tree).toMatchSnapshot();

});