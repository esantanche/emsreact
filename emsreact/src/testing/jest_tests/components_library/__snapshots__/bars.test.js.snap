// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MasterAppBar renders correctly 1`] = `
<header
  className="MuiPaper-root-10 MuiPaper-elevation4-16 MuiAppBar-root-2 MuiAppBar-positionSticky-5 MuiAppBar-colorPrimary-8 MasterAppBar-MasterAppBar-1"
/>
`;

exports[`MasterAvatar renders correctly 1`] = `
<header
  className="MuiPaper-root-10 MuiPaper-elevation4-16 MuiAppBar-root-2 MuiAppBar-positionSticky-5 MuiAppBar-colorPrimary-8 MasterAppBar-MasterAppBar-1"
/>
`;
